crypto/sm3.mbt

```diff
@@ -16,20 +16,21 @@
 // - [GM/T 0004-2012]    https://www.oscca.gov.cn/sca/xxgk/2010-12/17/1002389/files/302a3ada057c4a73830536d03e683110.pdf
 // SM3 is as secure as SHA256, providing similar performance. https://doi.org/10.3390/electronics8091033
 
-priv struct SM3Context {
+struct SM3Context {
   reg : FixedArray[UInt] // register A B C D E F G H. i.e. digest
   mut len : UInt64
-  mut msg : Array[Byte]
+  mut msg : Iter[Byte]
```

Comment:
Why save an `Iter`?

Reply:
currently `update()` `update_from_iter()` accepts `data` of type `Bytes` or `Iter[Byte]`. `data` is used to populate `msg`. and is further used in `sm3_transform`. Saving an Iter[Byte] means Bytes can be convert to Iter[Byte] at no cost, but not Array[Byte]